src/_tests/pages/workspaces/e2e/BatchAnalystFlow.test.js

```diff
@@ -0,0 +1,129 @@
+import expect from 'expect-puppeteer'
+import puppeteer from 'puppeteer'
+import { getAccessToken, wait, generateUUID, setupTest, cleanupTest, createWorkspace, waitForNetworkIdle } from 'src/_tests/test-utils'
+
+expect.setDefaultOptions({ timeout: 5555 })
+
+const DEBUG = false
+
+describe('Google', () => {
+  let browser
+  let page
+  let token
+  const workspaceName = 'WorkflowTestWS-'+ generateUUID()
+  const billingProjectName = 'general-dev-billing-account'
+
+  beforeEach(async () => [browser, page, token] = await setupTest(browser, page))
+
+  afterEach(async () => cleanupTest(browser, page, workspaceName, billingProjectName))
+
+
+  it('create a workspace and launch a batch analysis (workflow)', async () => {
+    await createWorkspace(page, workspaceName, billingProjectName)
+
+    // upload "small set" of data (& confirm)
+    await expect(page).toClick('[datatestid="data-tab"]')
+    await expect(page).toClick('[datatestid="uploadTsvButton"]')
+
+    const input = await page.$('input[type="file"]')
+    await input.uploadFile('./participant.tsv')
+    await expect(page).toClick('[datatestid="uploadButton"]')
+    await expect(page).toMatch('participant (1)')
+
+    await expect(page).toClick('[datatestid="tools-tab"]')
+    await expect(page).toMatch('No tools added')
+
+    // add method config
+    const workspaceUrl = page.url()
+    console.log('here')
+
+    await expect(page).toClick('[datatestid="navigationHamburgerIcon"]')
+    await expect(page).toClick('[datatestid="Code & Tools-link"]')
+    await expect(page).toClick('[datatestid="agoraLink"]')
+    await page.waitForNavigation({ waitUntil: 'networkidle0' }) // could wrap the following into a promise block with `page.waitForNavigation()`
+    // background signin for FireCloud
+    console.log('token used here is : '+token)
+    let executionContext = await page.mainFrame().executionContext()
```

Comment:
Maybe something like this?
```js
await page.mainFrame().executionContext()
  .then(xc => xc.evaluate(token => window.forceSignedIn(token), token))
```
Then we wouldn't have to instantiate `executionContext` or re-assign it below.

Reply:
Also, does `.evaluate(() => window.forceSignedIn(token))` not work?